Description: add a debian compat option
 debian policy for reasons beyond my apprehension wants a -T option to
 set the window title (like in xterm).
 to make debian's (and derivate's) life easier let's support a no-op
 -T option but hide it. this option will not appear in --help output
 but be accepted by the parser all the same.
 
 this enables Debian and friends to drop a long standing useless patch
 and at the same time prevents us from introducing a conflicting -T in
 the future.
Author: Harald Sitter <sitter@kde.org>
Origin: vendor
Bug: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=532029
Forwarded: https://git.reviewboard.kde.org/r/128299/
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
Index: konsole/src/main.cpp
===================================================================
--- konsole.orig/src/main.cpp
+++ konsole/src/main.cpp
@@ -292,6 +292,18 @@ void fillCommandLineOptions(QCommandLine
     parser.addPositionalArgument(QStringLiteral("[args]"),
                                  i18nc("@info:shell", "Arguments passed to command"));
 
+    // Add a no-op compatibility option to make Konsole compatible with
+    // Debian's policy on X terminal emulators.
+    // -T is technically meant to set a title, that is not really meaningful
+    // for Konsole as we have multiple user-facing options controlling
+    // the title and overriding whatever is set elsewhere.
+    // https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=532029
+    // https://www.debian.org/doc/debian-policy/ch-customized-programs.html#s11.8.3
+    auto titleOption = QCommandLineOption(QStringList() << QStringLiteral("T"),
+                                          QStringLiteral("Debian policy compatibility, not used"),
+                                          QStringLiteral("value"));
+    titleOption.setHidden(true);
+    parser.addOption(titleOption);
 }
 
 void fillAboutData(KAboutData& aboutData)
